api/service/version_endpoint_service.go

```diff
@@ -132,6 +138,20 @@ func (k *endpointService) DeployEndpoint(ctx context.Context, environment *model
 		return nil, err
 	}
 
+	// calling webhooks if there's any webhooks configured
+	if k.webhookManager != nil && k.webhookManager.IsEventConfigured(webhooks.OnModelVersionPredeployment) {
+		body := &webhooks.VersionEndpointRequest{
+			EventType:       webhooks.OnModelVersionPredeployment,
+			VersionEndpoint: endpoint,
+		}
+
+		err := k.webhookManager.InvokeWebhooks(ctx, webhooks.OnModelVersionPredeployment, body, webhookManager.NoOpCallback, webhookManager.NoOpErrorHandler)
+		if err != nil {
+			log.Errorf("unable to invoke webhooks for event type: %s, model: %s, version: %s, error: %v", webhooks.OnModelVersionDeployed, model.Name, version.ID, err)
```

Comment:
should it be `webhooks.OnModelVersionUndeployed`?

Reply:
Thank you far catching these two!